### ISSUE TITLE
Request for Asynchronous Feature Support

### ISSUE THREAD

**Issue by lichuangcheng** [slhck/ffmpeg-progress-yield#21]
It's great, is it considered to add support for async API? It could help programs to work in async environment without blocking the current thread.

This can be done with the help of asyncio's subprocess module, and the end result would look something like this:

```python
async def run(cmd):
    ff = FfmpegProgress(cmd)
    async for progress in ff.async_run_command_with_progress():
        print(f"{progress}/100")
```

**Comment by slhck** [comment]
Thanks for the suggestion. I haven't really worked with asyncio before. Could you maybe provide a pull request that shows what you mean? From what I understand it's probably just a small addition.

**Comment by slhck** [comment]
Please let me know if https://github.com/slhck/ffmpeg-progress-yield/pull/22 does what you need!

**Comment by lichuangcheng** [comment]
Yes!!! That's what I want! It works better than my own messed up alterations. Thank you very, very, very much!

**Comment by slhck** [comment]
Great. I will merge after fixing the type checks.

**Comment by slhck** [comment]
Released as v0.8.0.
